lib/partial/transmuxer.js

```diff
@@ -134,27 +134,35 @@ var tsPipeline = function(options) {
     });
   });
 
-  /* TODO caption support
-   * Caption support may be trickier than for full segment appends, as the captions can
-   * be split and processed in a much more granular fashion.
-  pipeline.captionStream.on('data', function(data) {
-    pendingCaptions.push(data);
-  });
+  pipeline.captionStream.on('data', function(caption) {
+    var timelineStartPts;
+    var startTime;
+    var endTime;
+
+    if (pipeline.tracks.video) {
+      timelineStartPts = pipeline.tracks.video.timelineStartInfo.pts || 0;
+    } else {
+      // TODO: how do we handle the case where captions are returned before
+      // determining timelineStartPts?
+      timelineStartPts = 0;
+    }
 
-  pipeline.captionStream.on('flush', function() {
     // Translate caption PTS times into second offsets into the
-    // video timeline for the segment, and add track info
-    for (i = 0; i < this.pendingCaptions.length; i++) {
-      caption = this.pendingCaptions[i];
-      caption.startTime = (caption.startPts - timelineStartPts);
-      caption.startTime /= 90e3;
-      caption.endTime = (caption.endPts - timelineStartPts);
-      caption.endTime /= 90e3;
-      event.captionStreams[caption.stream] = true;
-      event.captions.push(caption);
+    // video timeline for the segment
+    startTime = caption.startPts;
```

Comment:
Is this the same logic as is in `adjustCaptionTiming`?

Reply:
This is pretty similar. I was thinking of moving adjustCaptionTiming and adjustMetadataTiming to a util file. Thoughts?